spec/lib/newspaper_works/data/work_derivatives_spec.rb

```diff
@@ -0,0 +1,193 @@
+# encoding: UTF-8
+
+require 'spec_helper'
+require 'misc_shared'
+
+RSpec.describe NewspaperWorks::Data::WorkDerivatives do
+  include_context "shared setup"
+
+  let(:bare_work) do
+    work = NewspaperPage.new
+    work.title = ['Another one']
+    work.save!
+    work
+  end
+
+  let(:work) do
+    # sample work comes from shared setup, but we need derivative, save...
+    mk_txt_derivative(sample_work)
+    sample_work.save!(validate: false)
+    sample_work
+  end
+
+  let(:txt1) do
+    file = Tempfile.new('txt1.txt')
+    file.write('hello')
+    file.close
+    file.path
+  end
+
+  let(:txt2) do
+    file = Tempfile.new('txt2.txt')
+    file.write('bye')
+    file.close
+    file.path
+  end
+
+  let(:encoded_text) do
+    file = Tempfile.new('txt_encoded.txt', encoding: 'UTF-8')
+    file.write('Gorgonzola Dolce® — on sale for £12.50/kg')
+    file.close
+    file.path
+  end
+
+  describe "enumerates available derivatives like hash" do
+    it "includes expected derivative path for work" do
+      adapter = described_class.new(work)
```

Comment:
Is there a way to DRY this line:
```ruby
adapter = described_class.new(work)
```
(It's used in most of the specs, it seems? Granted there are some permutations here and there.)

Or does `adapter` get mutated by some of the actions happening in the various specs (e.g. `mk_jp2_derivative`)?

Reply:
Probably a `let` statement for all cases where `work` is being adapted, and not use of `bare_work`.  That would DRY enough to justify doing it.